api/src/pcapi/utils/date.py

```diff
@@ -107,6 +107,11 @@ def isoformat(date_to_format: datetime | date) -> str:
     return date_to_format.isoformat()
 
 
+def format_to_second_date(date_to_format: datetime | date) -> str:
+    remove_milisecond = 19
+    return date_to_format.isoformat()[:remove_milisecond]
```

Comment:
```suggestion
    if isinstance(date_to_format, date):
       return date_to_format.isoformat()
    return date_to_format.isoformat(timespecs="seconds")
```

https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat : )

Reply:
En plus je n’ai pas l'impression que tu lui passes une date à un moment donné, c’est toujours un datetime non ?

Reply:
Le but de la fonction est justement de virer les milisecondes, si je fais ca et bien pour les dates entré cela ne vire pas les milisecondes et les tests fail. 
quand a si je vire la condition pour le timespecs, et bien tout plante tout simplement.
Du coup je ne comprend pas ce changement :/

Reply:
Les millisecondes sont bien enlevées non ?

```python

development❯ datetime.utcnow().isoformat()
'2024-01-09T09:54:25.538385'

development❯ datetime.utcnow().isoformat(timespec="seconds")
'2024-01-09T09:54:26'

Reply:
Ce que propose Damien m'a pourtant l'air de bien fonctionner : 

```python
>>> datetime.datetime.utcnow().isoformat()
'2024-01-09T09:56:28.211847'
>>> datetime.datetime.utcnow().isoformat(timespec="seconds")
'2024-01-09T09:56:29'
```

... ceci dit, est-ce qu'on a vraiment besoin de définir une fonction qui serait utilisée seulement dans les tests ? Est-ce que les tests ne pourraient pas utiliser la fonction de formatage de date utilisée par l'implémentation ?

Reply:
Et bien prenons par exemple le resultat de stock.beginningDateTime dans test_get_offer du fichier get_collective_offer_test: 
RESULTAT : 2024-01-10 10:13:42.226945
RESULTAT APRES FONCTION : '2024-01-10T10:13:42.226945' ( on voit toujours le 226945 )
sur ce cas la fonction passe par un if donc ne va pas vers le timespec. 
Dans le cas ou la fonction va dans le timespec et bien cela ne fonctionne pas avec les objets de type date ( étant donné le if )

Reply:
J’ai trouvé en testant 5 minutes dans mon shell python:

```python
development❯ from datetime import datetime, date
development❯ isinstance(datetime.utcnow(), date)
True
development❯ isinstance(datetime.utcnow().date(), date)
True
development❯ isinstance(datetime.utcnow().date(), datetime)
False
```

une `datetime` est une instance de `date`, mais la réciproque n’est pas vraie
du coup tu peux juste inverser la condition de ma proposition:

```
    if isinstance(date_to_format, datetime):
       return date_to_format.isoformat(timespecs="seconds")
    return date_to_format.isoformat() 
```

Reply:
ce qui est logique finalement, une date composée d'une année, d'un mois, d'un jour, de l'heure, des secondes et des millisecondes, est une date malgré tout.
La réciproque n'est pas vraie, le 1er janvier 1970 est différent du 1er janvier 1970 à 00:00.0000
Je penses que tu aurais pu trouver tout seul ;)

Reply:
oohh je vois c'est juste que du coup j'imaginai plus complexe qu'une simple inversion x)